shakenfist/config.py

```diff
@@ -111,7 +111,14 @@ def __init__(self):
         for var in os.environ:
             if var.startswith('SHAKENFIST_'):
                 flag = var.replace('SHAKENFIST_', '')
-                self.config[flag] = os.environ[var]
+                value = os.environ[var]
+
+                try:
+                    value = float(value)
```

Comment:
How will this handle non floating values? eg. API_PORT

Reply:
They'll get a valueerror, and fall through to using the original value. For example:

```
$ python3
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> foo="1"
>>> try:
...   foo=float(foo)
... except ValueError:
...   pass
... 
>>> print(foo)
1.0
>>> foo="gerkin"
>>> try:
...   foo=float(foo)
... except ValueError:
...   pass
... 
>>> print(foo)
gerkin
```

Reply:
I understand the processing for text/numerics.

I should have been a little clearer. Integer values such API_PORT are now floating-point values - as you actually demonstrated above. This breaks other parts of the code. Some parts still work because they are formatting with %d but others are using %s such as the gunicorn API_TIMEOUT.

In the future the value may be directly applied and an integer is expected.

Maybe test for an integer first?

Reply:
Oh right. I see now. Its interesting. I wonder if we need a type system for configuration values? I guess I want to avoid that complexity if possible.

Reply:
The problem being, that a string of "123" is a valid int and float. As is "123.123". Both casts will work and not raise a ValueError.

Reply:
If you keep the same logic, I was thinking of something simple like
```
try:
    foo = int(foo)
except:
    try:
        foo=float(foo)
    except ValueError:
        pass
```